RandLAPACK/drivers/rl_hqrrp.hh

```diff
@@ -0,0 +1,873 @@
+void _LAPACK_lafrb(
+    lapack::Side side,
+    lapack::Op op,
+    lapack::Direction dir,
+    lapack::StoreV  storev,
+    int64_t m, int64_t n, int64_t k,
+    T *buff_U, int64_t ldu,
+    T *buff_T, int64_t ldt,
+    T *buff_B, int64_t ldb,
+    T *buff_W, int64_t ldw
+){
+    lapack_int m_ = (lapack_int) m;
+    lapack_int n_ = (lapack_int) n;
+    lapack_int k_ = (lapack_int) k;
+    lapack_int ldim_U = (lapack_int) ldu;
+    lapack_int ldim_T = (lapack_int) ldt;
+    lapack_int ldim_B = (lapack_int) ldb;
+    lapack_int ldim_W = (lapack_int) ldw;
+    char side_ = blas::side2char( side );
+    char trans_ = blas::op2char( op );
+    char direction_ = lapack::direction2char( dir );
+    char storev_ = lapack::storev2char( storev );
+    if (typeid(T) == typeid(double)) {
+        LAPACK_dlarfb( & side_, & trans_, & direction_, & storev_,  
+                    & m_, & n_, & k_, (double *) buff_U, & ldim_U, (double *) buff_T, & ldim_T, 
+                    (double *) buff_B, & ldim_B, (double *) buff_W, & ldim_W
+                    #ifdef LAPACK_FORTRAN_STRLEN_END
+                    , 1, 1, 1, 1
+                    #endif
+                    );
+    } else if (typeid(T) == typeid(float)) {
+        LAPACK_slarfb( & side_, & trans_, & direction_, & storev_,  
+                    & m_, & n_, & k_, (float *) buff_U, & ldim_U, (float *) buff_T, & ldim_T, 
+                    (float *) buff_B, & ldim_B, (float *) buff_W, & ldim_W
+                    #ifdef LAPACK_FORTRAN_STRLEN_END
+                    , 1, 1, 1, 1
+                    #endif
+                    );
+    } else {
+        // Unsupported type
+    }
+    return;
+}
+
+// ============================================================================
+template <typename T>
+void _LAPACK_larf(
+    lapack::Side side,
+    int64_t m, int64_t n,
+    T *v, int64_t inc_v, T* tau,
+    T *C, int64_t ldc,
+    T *work
+){
+    char side_ = blas::side2char( side );
+    lapack_int inc_v_ = (lapack_int) inc_v;
+    lapack_int m_ = (lapack_int) m;
+    lapack_int n_ = (lapack_int) n;
+    lapack_int ldc_ = (lapack_int) ldc;
+    if (typeid(T) == typeid(double)) {
+        LAPACK_dlarf( & side_, & m_, & n_, 
+            (double *) v, & inc_v_,
+            (double *) tau,
+            (double *) C, & ldc_,
+            (double *) work
+            #ifdef LAPACK_FORTRAN_STRLEN_END
+            , 1
+            #endif
+            );
+    } else if (typeid(T) == typeid(float)) {
+        LAPACK_slarf( & side_, & m_, & n_, 
+            (float *) v, & inc_v_,
+            (float *) tau,
+            (float *) C, & ldc_,
+            (float *) work
+            #ifdef LAPACK_FORTRAN_STRLEN_END
+            , 1
+            #endif
+            );
+    } else {
+        // Unsupported type
+    }
+    return;
+}
+
+/* ========================================================================= */
+template <typename T>
+static T NoFLA_Normal_random_number( T mu, T sigma ) {
```

Comment:
It's important to note that this function handles all randomness in the HQRRP implementation. That is to say -- RandBLAS ``RNGState`` objects can't be used to specify the state of the random number generator. At the very least there should be a TODO here to avoid relying on this function.

Reply:
So then, isnt everything alright here?

Reply:
Resolved in the comment below.